### ISSUE TITLE
xapian_quilt.tar.gz missing in rc1 tarball

### ISSUE THREAD

**Issue by hagedose** [cyrusimap/cyrus-imapd#1802]
According to README.xapian, the contrib/ dir should contain xapian_quilt.tar.gz. It is missing in cyrus-imapd-3.0.0-rc1.tar.gz.

**Comment by brong** [comment]
README.xapian should be updated with the latest state of how to build xapian from the git repository.  We're not patching with a quilt series any more.

**Comment by rsto** [comment]
Yep, I'm onto it.

**Comment by rsto** [comment]
Done. https://github.com/cyrusimap/cyrus-imapd/commit/809a6acd152e7bc61a355d563d9d72ca94d757c8 @hagedose please let me know if there's anything unclear, I'll be happy to update the instructions

**Comment by hagedose** [comment]
Thanks. It took several attempts because I had to install a number of additional dependencies on CentOS 7, but then build.sh worked. I wonder if it shouldn't be included in the tarball? If you're building from git, it makes sense to pull in the cyruslibs from git as well, but when you're working with a tarball, you'd generally expect to have everything included that you need.

**Comment by rsto** [comment]
Ack. I don't think we haven't decided yet how to bundle 3.0 tarballs. I'll leave this issue open until we do.

**Comment by rsto** [comment]
The release process doc draft now contains a section how to bundle our Xapian fork -> https://github.com/cyrusimap/cyrus-imapd/commit/e1f8e7a5e3cadb21dfacd970030736ca3b79224a

**Comment by elliefm** [comment]
Running `make distcheck` in the top level of the xapian repository eventually produces the following tarballs:

```
$find . -name \*.tar.xz
./xapian-applications/omega/xapian-omega-1.4.0.tar.xz
./xapian-bindings/xapian-bindings-1.4.0.tar.xz
./xapian-core/xapian-core-1.4.0.tar.xz
```

Do we know if Cyrus needs any of these other than xapian-core?  About to log off for the day, but will experiment more tomorrow.

**Comment by rsto** [comment]
No, we only need xapian-core (see step 3 in the release doc).

**Comment by elliefm** [comment]
Yeah I saw that in the release doc, but conversely, build.sh from cyruslibs seems to install the whole lot, so I was double checking :)

**Comment by rsto** [comment]
@elliefm I'm tempted to change build.sh to only build xapian-core (that's what I am doing on my machine). It's faster to build and has less dependencies. Or are we using any xapian module other than core outside Cyrus?

**Comment by brong** [comment]
I have already done this, hang on a sec...



   ./configure --enable-silent-rules --prefix=$PREFIX

+  cd xapian-core

   make $MAKEOPTS



that's all it takes :)





On Wed, 15 Feb 2017, at 16:54, Robert Stepanek wrote:

> @elliefm[1] I'm tempted to change build.sh to only build xapian-core
> (that's what I am doing on my machine). It's faster to build and has
> less dependencies. Or are we using any xapian module other than core
> outside Cyrus?
> — You are receiving this because you commented. Reply to this email
> directly, view it on GitHub[2], or mute the thread[3].
> 



--

  Bron Gondwana

  brong@fastmail.fm






Links:

  1. https://github.com/elliefm
  2. https://github.com/cyrusimap/cyrus-imapd/issues/1802#issuecomment-279923810

**Comment by elliefm** [comment]
So I've done:

* clone our fork of the Xapian repository
* bootstrap, configure, make, make distcheck (results in a `xapian-core/xapian-core-1.4.0.tar.xz`)
* unpack the tarball somewhere
* build and install from the tarball source
* configure Cyrus with `XAPIAN_CONFIG=</path/to/xapian-config> --enable-xapian`

and now building Cyrus, I'm getting these errors from xapian_wrap.cpp:

```
imap/xapian_wrap.cpp: In function ‘int xapian_dbw_open(const char**, xapian_dbw_t**)’:
imap/xapian_wrap.cpp:259:40: error: ‘FLAG_CJK_WORDS’ is not a member of ‘Xapian::TermGenerator’
         dbw->term_generator->set_flags(Xapian::TermGenerator::FLAG_CJK_WORDS,
                                        ^~~~~~
imap/xapian_wrap.cpp:260:18: error: ‘FLAG_CJK_WORDS’ is not a member of ‘Xapian::TermGenerator’
                 ~Xapian::TermGenerator::FLAG_CJK_WORDS);
                  ^~~~~~
imap/xapian_wrap.cpp: In function ‘xapian_query_t* xapian_query_new_match_internal(const xapian_db_t*, int, int, const char*)’:
imap/xapian_wrap.cpp:559:37: error: ‘FLAG_CJK_WORDS’ is not a member of ‘Xapian::QueryParser’
                                     Xapian::QueryParser::FLAG_CJK_WORDS,
                                     ^~~~~~
  CC       sieve/sieve_libcyrus_sieve_la-comparator.lo
imap/xapian_wrap.cpp: In function ‘Xapian::Query xapian_snipgen_build_query(xapian_snipgen_t*)’:
imap/xapian_wrap.cpp:827:30: error: ‘FLAG_CJK_WORDS’ is not a member of ‘Xapian::TermGenerator’
     term_generator.set_flags(Xapian::TermGenerator::FLAG_CJK_WORDS,
                              ^~~~~~
imap/xapian_wrap.cpp:828:14: error: ‘FLAG_CJK_WORDS’ is not a member of ‘Xapian::TermGenerator’
             ~Xapian::TermGenerator::FLAG_CJK_WORDS);
              ^~~~~~
imap/xapian_wrap.cpp: In function ‘int xapian_snipgen_end_doc(xapian_snipgen_t*, buf*)’:
imap/xapian_wrap.cpp:908:17: error: ‘SNIPPET_EMPTY_WITHOUT_MATCH’ is not a member of ‘Xapian::MSet’
                 Xapian::MSet::SNIPPET_EMPTY_WITHOUT_MATCH|
                 ^~~~~~
imap/xapian_wrap.cpp:913:17: error: ‘FLAG_CJK_WORDS’ is not a member of ‘Xapian::TermGenerator’
                 Xapian::TermGenerator::FLAG_CJK_WORDS);
                 ^~~~~~
  CC       sieve/sieve_libcyrus_sieve_la-flags.lo
Makefile:5671: recipe for target 'imap/imap_libcyrus_imap_la-xapian_wrap.lo' failed
```

Are those errors familiar?  Suspect I've missed something obvoius

**Comment by rsto** [comment]
Are you sure you updated your xapian git submodule lately? Yours builds 1.4.0 but it should 1.5.0...

    $ git submodule
    4d3d02ba6ee92d4a9764ad1c98c8004e76dbf06a xapian (v1.3.6-475-g4d3d02ba6ee9)

and 

    xapian/xapian-core$ git log | head -n 10
    commit 4d3d02ba6ee92d4a9764ad1c98c8004e76dbf06a 
    Author: Bron Gondwana <brong@fastmail.fm>
    Date:   Fri Jun 24 12:27:07 2016 +1000

    bootstrap: handle being a git submodule

    commit 2843cb3869de851d5eeaf3697eee90626cc40fe2
    Author: Robert Stepanek <rsto@paranoia.at>
    Date:   Thu Jan 26 13:48:47 2017 +0100

**Comment by rsto** [comment]
Also: make sure

    XAPIAN_CONFIG="$CYRUSLIBS/bin/xapian-config-1.5"

when you run cyrus-imapd's `./configure`

**Comment by elliefm** [comment]
I'm not using a submodule (I don't use cyruslibs), I just forked straight from cyrusimap/xapian...

Oh I see, there's a master branch and a cyrus branch there -- and I was building the master branch, not the cyrus branch.  Cool, will throw it away and try again properly tomorrow, thanks :)

**Comment by hagedose** [comment]
Maybe I'm confused, but after the changes I expected cyrus-imapd-3.0.0-rc3.tar.gz to contain cyruslibs?!

What's more with the cyruslibs I built previously, the compile (which worked in rc1 and rc2) now fails:

```
make[2]: Entering directory `/root/cyrus-imapd-3.0.0-rc3'
/bin/sh ./libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.   -I. -I./lib -I. -I./lib -DLIBEXEC_DIR=\"/usr/libexec\" -DSBIN_DIR=\"/usr/sbin\" -DSYSCONF_DIR=\"/etc\" -DHAVE_CONFIG_H         -I/usr/local/cyruslibs/include -g -O2 -MT imap/imap_libcyrus_imap_la-xapian_wrap.lo -MD -MP -MF imap/.deps/imap_libcyrus_imap_la-xapian_wrap.Tpo -c -o imap/imap_libcyrus_imap_la-xapian_wrap.lo `test -f 'imap/xapian_wrap.cpp' || echo './'`imap/xapian_wrap.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I. -I./lib -I. -I./lib -DLIBEXEC_DIR=\"/usr/libexec\" -DSBIN_DIR=\"/usr/sbin\" -DSYSCONF_DIR=\"/etc\" -DHAVE_CONFIG_H -I/usr/local/cyruslibs/include -g -O2 -MT imap/imap_libcyrus_imap_la-xapian_wrap.lo -MD -MP -MF imap/.deps/imap_libcyrus_imap_la-xapian_wrap.Tpo -c imap/xapian_wrap.cpp  -fPIC -DPIC -o imap/.libs/imap_libcyrus_imap_la-xapian_wrap.o
imap/xapian_wrap.cpp: In Funktion »int xapian_snipgen_end_doc(xapian_snipgen_t*, buf*)«:
imap/xapian_wrap.cpp:908:17: Fehler: »SNIPPET_EMPTY_WITHOUT_MATCH« ist kein Element von »Xapian::MSet«
                 Xapian::MSet::SNIPPET_EMPTY_WITHOUT_MATCH|
                 ^
make[2]: *** [imap/imap_libcyrus_imap_la-xapian_wrap.lo] Fehler 1
make[2]: Leaving directory `/root/cyrus-imapd-3.0.0-rc3'
make[1]: *** [all-recursive] Fehler 1
make[1]: Leaving directory `/root/cyrus-imapd-3.0.0-rc3'
```

Any ideas?

**Comment by elliefm** [comment]
Still working out if/how I want to bundle them -- cyruslibs/xapian are not included in rc3 yet (hence no mention of them in the release notes :) )

I'm not sure what that error you're getting is, but maybe you need to fetch/rebuild xapian again.  I think maybe it got updated for changes that occurred just after rc2 dropped (which are now in rc3)

**Comment by hagedose** [comment]
Thanks. It might take me a few days to get to that.

**Comment by elliefm** [comment]
No worries, and good luck!

We're expecting to get our patches into xapian proper at some point, but I don't have a clear idea of how long that might take -- so hopefully at some point it will just be "install xapian from a standard release", and this whole issue will go away. :)

**Comment by hagedose** [comment]
That worked, but 809a6ac needs to be fixed to reference

 `XAPIAN_CONFIG="$CYRUSLIBS/bin/xapian-config-1.5`

instead.

**Comment by elliefm** [comment]
Cheers, fixed in 607b996 :)
